File: solutions/kotlin/raindrops/1/src/main/kotlin/Raindrops.kt
object Raindrops {
    private val NAMED_FACTORS = mapOf(3 to "Pling", 5 to "Plang", 7 to "Plong")

    fun convert(drops: Int) =
        NAMED_FACTORS
            .filterKeys { factor -> drops % factor == 0 }
            .values
            .joinToString("")
            .let { // Can't use ifEmpty, because kotlin version is set to 1.2.40, and it's 1.3
                if (it.isEmpty()) drops.toString() else it
            }
}

File: solutions/javascript/raindrops/1/raindrops.js
/* eslint-disable linebreak-style */
export default class Raindrops {
  convert(count) {
    return Raindrops.pling(count) + Raindrops.plang(count) + Raindrops.plong(count) || `${count}`;
  }

  static pling(count) {
    return (!(count % 3) && 'Pling') || '';
  }

  static plang(count) {
    return (!(count % 5) && 'Plang') || '';
  }

  static plong(count) {
    return (!(count % 7) && 'Plong') || '';
  }
}
